### DISCUSSION TITLE
How to set the uid:gid of the server process and data files to prevent permission denied errors.

### DISCUSSION THREAD

**Discussion by Tatonker** [felddy/foundryvtt-docker#160] [Q&A] [answered]
Hi,

I'm using the container to run Foundry, and when I try to copy files from my Windows PC to my /foundrydata/Data folder on my server get a permission denied error as follows:

![image](https://user-images.githubusercontent.com/70360365/111886171-74522f00-8989-11eb-8872-d672e932f248.png)

In WinSCP I can see that the the owner of the Foundry Data folder is listed as 421 and that the rights are rwx-r-x-r-x

![image](https://user-images.githubusercontent.com/70360365/111886196-a95e8180-8989-11eb-98f3-5359480e4986.png)

I've tried to identify this user id in my server and I can't find it. Can anyone help? I'm very new to Linux so any help is appreciated.

Regards,

Tatonker

Edit: I'll try the answer posted in #153 and feedback.

**Comment by felddy** [comment, marked as answer]
#153 should be a path to success by disabling permission normalization, but you may run into other permission errors if you aren't vigilant.

If you know the `uid` and `gid` that you want those directories to have you can set the `FOUNDRY_UID` and/or `FOUNDRY_GID` environment variables.  See the README for details: https://github.com/felddy/foundryvtt-docker#optional

**Reply by Tatonker** [reply]
You sir are a legend. I knew the uid and gid so put those into my compose file, re-ran the compose file and now have been able to successfully transfer over 500MB of assets from my windows laptop to my Foundry instance. Not being able to transfer files has been bugging me for about 9 months but never bothered to try to figure it out until last night when I subbed to the JB2A animated token patreon. Thank you so much for your help, I really appreciate it.
